File: apps/lib/services/accessibility_service.dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';
import 'package:shared_preferences/shared_preferences.dart';

/// WCAG 2.2 AAA Accessibility Service
///
/// Provides accessibility utilities for Bay Navigator including:
/// - Reduced motion detection and preference
/// - High contrast mode support
/// - Screen reader announcements
/// - Text customization (WCAG 3.0 draft)
/// - Semantic label helpers
class AccessibilityService {
  static const String _reduceMotionKey = 'accessibility_reduce_motion';
  static const String _highContrastKey = 'accessibility_high_contrast';
  static const String _textScaleKey = 'accessibility_text_scale';
  static const String _lineHeightKey = 'accessibility_line_height';
  static const String _letterSpacingKey = 'accessibility_letter_spacing';
  static const String _wordSpacingKey = 'accessibility_word_spacing';

  /// Check if reduced motion is preferred (system or user setting)
  static bool shouldReduceMotion(BuildContext context) {
    final mediaQuery = MediaQuery.of(context);
    return mediaQuery.disableAnimations;
  }

  /// Check if high contrast is preferred
  static bool shouldUseHighContrast(BuildContext context) {
    final mediaQuery = MediaQuery.of(context);
    return mediaQuery.highContrast;
  }

  /// Check if bold text is enabled
  static bool shouldUseBoldText(BuildContext context) {
    final mediaQuery = MediaQuery.of(context);
    return mediaQuery.boldText;
  }

  /// Get system text scale factor
  static double getTextScaleFactor(BuildContext context) {
    return MediaQuery.textScalerOf(context).scale(1.0);
  }

  /// Announce a message to screen readers
  static Future<void> announce(String message, {TextDirection textDirection = TextDirection.ltr}) async {
    await SemanticsService.announce(message, textDirection);
  }

  /// Announce with polite priority (doesn't interrupt current speech)
  static Future<void> announcePolitely(String message) async {
    await announce(message);
  }

  /// Announce with assertive priority (interrupts current speech)
  static Future<void> announceAssertively(String message) async {
    // In Flutter, all announcements are assertive by default
    await announce(message);
  }

  /// Announce loading state
  static Future<void> announceLoading() async {
    await announce('Loading');
  }

  /// Announce loading complete
  static Future<void> announceLoadingComplete([String? itemCount]) async {
    if (itemCount != null) {
      await announce('Loaded $itemCount items');
    } else {
      await announce('Loading complete');
    }
  }

  /// Announce error
  static Future<void> announceError(String error) async {
    await announce('Error: $error');
  }

  /// Announce filter change
  static Future<void> announceFilterChange(int resultCount) async {
    await announce('Showing $resultCount results');
  }

  /// Announce save/unsave action
  static Future<void> announceSaveAction(bool isSaved, String programName) async {
    if (isSaved) {
      await announce('$programName saved to your list');
    } else {
      await announce('$programName removed from your list');
    }
  }

  /// Get animation duration respecting reduced motion
  static Duration getAnimationDuration(BuildContext context, Duration normalDuration) {
    if (shouldReduceMotion(context)) {
      return Duration.zero;
    }
    return normalDuration;
  }

  /// Get animation curve respecting reduced motion
  static Curve getAnimationCurve(BuildContext context, Curve normalCurve) {
    if (shouldReduceMotion(context)) {
      return Curves.linear;
    }
    return normalCurve;
  }

  /// Save text customization settings
  static Future<void> saveTextSettings({
    double? textScale,
    double? lineHeight,
    double? letterSpacing,
    double? wordSpacing,
  }) async {
    final prefs = await SharedPreferences.getInstance();
    if (textScale != null) {
      await prefs.setDouble(_textScaleKey, textScale);
    }
    if (lineHeight != null) {
      await prefs.setDouble(_lineHeightKey, lineHeight);
    }
    if (letterSpacing != null) {
      await prefs.setDouble(_letterSpacingKey, letterSpacing);
    }
    if (wordSpacing != null) {
      await prefs.setDouble(_wordSpacingKey, wordSpacing);
    }
  }

  /// Load text customization settings
  static Future<TextSettings> loadTextSettings() async {
    final prefs = await SharedPreferences.getInstance();
    return TextSettings(
      textScale: prefs.getDouble(_textScaleKey) ?? 1.0,
      lineHeight: prefs.getDouble(_lineHeightKey) ?? 1.5,
      letterSpacing: prefs.getDouble(_letterSpacingKey) ?? 0.0,
      wordSpacing: prefs.getDouble(_wordSpacingKey) ?? 0.0,
    );
  }

  /// Reset text settings to defaults
  static Future<void> resetTextSettings() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove(_textScaleKey);
    await prefs.remove(_lineHeightKey);
    await prefs.remove(_letterSpacingKey);
    await prefs.remove(_wordSpacingKey);
  }
}

/// Text customization settings (WCAG 3.0 draft compliance)
class TextSettings {
  final double textScale;      // 0.8 - 1.5
  final double lineHeight;     // 1.5 - 2.5
  final double letterSpacing;  // 0 - 0.1em
  final double wordSpacing;    // 0 - 0.2em

  const TextSettings({
    this.textScale = 1.0,
    this.lineHeight = 1.5,
    this.letterSpacing = 0.0,
    this.wordSpacing = 0.0,
  });

  TextSettings copyWith({
    double? textScale,
    double? lineHeight,
    double? letterSpacing,
    double? wordSpacing,
  }) {
    return TextSettings(
      textScale: textScale ?? this.textScale,
      lineHeight: lineHeight ?? this.lineHeight,
      letterSpacing: letterSpacing ?? this.letterSpacing,
      wordSpacing: wordSpacing ?? this.wordSpacing,
    );
  }

  /// Apply text settings to a TextStyle
  TextStyle applyTo(TextStyle style) {
    return style.copyWith(
      height: lineHeight,
      letterSpacing: letterSpacing,
      wordSpacing: wordSpacing,
    );
  }

  bool get isDefault =>
      textScale == 1.0 &&
      lineHeight == 1.5 &&
      letterSpacing == 0.0 &&
      wordSpacing == 0.0;
}

/// High contrast color overrides
class HighContrastColors {
  // Enhanced contrast colors (10:1+ ratio)
  static const Color text = Color(0xFF000000);
  static const Color textOnDark = Color(0xFFFFFFFF);
  static const Color background = Color(0xFFFFFFFF);
  static const Color backgroundDark = Color(0xFF000000);
  static const Color primary = Color(0xFF004040); // Darker teal for 10:1
  static const Color primaryOnDark = Color(0xFF80FFFF); // Bright cyan
  static const Color link = Color(0xFF0000CC); // Traditional link blue
  static const Color linkOnDark = Color(0xFF99CCFF);
  static const Color focus = Color(0xFF000000);
  static const Color focusOnDark = Color(0xFFFFFF00); // Yellow focus ring
  static const Color error = Color(0xFFCC0000);
  static const Color errorOnDark = Color(0xFFFF6666);
  static const Color success = Color(0xFF006600);
  static const Color successOnDark = Color(0xFF66FF66);
}

/// Semantic label helpers for consistent accessibility descriptions
class SemanticLabels {
  // Navigation
  static const String home = 'Home';
  static const String directory = 'Program Directory';
  static const String saved = 'Saved Programs';
  static const String settings = 'Settings';
  static const String search = 'Search programs';
  static const String filter = 'Filter programs';
  static const String sort = 'Sort programs';

  // Actions
  static String saveProgram(String name) => 'Save $name to your list';
  static String unsaveProgram(String name) => 'Remove $name from your list';
  static String shareProgram(String name) => 'Share $name';
  static String callPhone(String phone) => 'Call $phone';
  static String openWebsite(String name) => 'Open $name website';
  static String getDirections(String location) => 'Get directions to $location';

  // States
  static String programSaved(String name) => '$name is saved';
  static String programNotSaved(String name) => '$name is not saved';
  static String filterActive(int count) => '$count filters active';
  static String resultsCount(int count) => '$count programs found';
  static String loadingPrograms = 'Loading programs';

  // Categories
  static String categoryLabel(String category) => '$category programs';

  // Groups
  static const Map<String, String> groupLabels = {
    'seniors': 'Seniors 65 and older',
    'veterans': 'Veterans and military families',
    'disabled': 'People with disabilities',
    'lowincome': 'Low income individuals and families',
    'students': 'Students',
    'families': 'Families with children',
    'unemployed': 'Unemployed or job seekers',
    'homeless': 'People experiencing homelessness',
    'immigrants': 'Immigrants and refugees',
    'youth': 'Youth and young adults',
  };

  static String getGroupLabel(String groupId) {
    return groupLabels[groupId.toLowerCase()] ?? groupId;
  }
}

/// Widget wrapper for accessible animations
class AccessibleAnimatedContainer extends StatelessWidget {
  final Widget child;
  final Duration duration;
  final Curve curve;
  final AlignmentGeometry? alignment;
  final EdgeInsetsGeometry? padding;
  final Color? color;
  final Decoration? decoration;
  final double? width;
  final double? height;
  final BoxConstraints? constraints;
  final Matrix4? transform;

  const AccessibleAnimatedContainer({
    super.key,
    required this.child,
    this.duration = const Duration(milliseconds: 300),
    this.curve = Curves.easeInOut,
    this.alignment,
    this.padding,
    this.color,
    this.decoration,
    this.width,
    this.height,
    this.constraints,
    this.transform,
  });

  @override
  Widget build(BuildContext context) {
    final actualDuration = AccessibilityService.getAnimationDuration(context, duration);
    final actualCurve = AccessibilityService.getAnimationCurve(context, curve);

    return AnimatedContainer(
      duration: actualDuration,
      curve: actualCurve,
      alignment: alignment,
      padding: padding,
      color: color,
      decoration: decoration,
      width: width,
      height: height,
      constraints: constraints,
      transform: transform,
      child: child,
    );
  }
}

/// Widget wrapper for accessible opacity transitions
class AccessibleAnimatedOpacity extends StatelessWidget {
  final Widget child;
  final double opacity;
  final Duration duration;
  final Curve curve;

  const AccessibleAnimatedOpacity({
    super.key,
    required this.child,
    required this.opacity,
    this.duration = const Duration(milliseconds: 300),
    this.curve = Curves.easeInOut,
  });

  @override
  Widget build(BuildContext context) {
    if (AccessibilityService.shouldReduceMotion(context)) {
      return Opacity(opacity: opacity, child: child);
    }

    return AnimatedOpacity(
      duration: duration,
      curve: curve,
      opacity: opacity,
      child: child,
    );
  }
}

/// Widget wrapper for accessible scale transitions
class AccessibleAnimatedScale extends StatelessWidget {
  final Widget child;
  final double scale;
  final Duration duration;
  final Curve curve;

  const AccessibleAnimatedScale({
    super.key,
    required this.child,
    required this.scale,
    this.duration = const Duration(milliseconds: 300),
    this.curve = Curves.easeInOut,
  });

  @override
  Widget build(BuildContext context) {
    if (AccessibilityService.shouldReduceMotion(context)) {
      return Transform.scale(scale: scale, child: child);
    }

    return AnimatedScale(
      duration: duration,
      curve: curve,
      scale: scale,
      child: child,
    );
  }
}

/// Extension to add accessible semantics to widgets easily
extension AccessibleWidget on Widget {
  /// Wrap widget with semantic label
  Widget withSemanticLabel(String label) {
    return Semantics(
      label: label,
      child: this,
    );
  }

  /// Wrap widget as a button with label
  Widget asAccessibleButton({
    required String label,
    String? hint,
    VoidCallback? onTap,
  }) {
    return Semantics(
      button: true,
      label: label,
      hint: hint,
      onTap: onTap,
      child: this,
    );
  }

  /// Wrap widget as a heading
  Widget asAccessibleHeading(String label) {
    return Semantics(
      header: true,
      label: label,
      child: this,
    );
  }

  /// Wrap widget as an image with description
  Widget asAccessibleImage(String description) {
    return Semantics(
      image: true,
      label: description,
      child: this,
    );
  }

  /// Wrap widget as a link
  Widget asAccessibleLink(String label) {
    return Semantics(
      link: true,
      label: label,
      child: this,
    );
  }

  /// Wrap widget to exclude from semantics (decorative)
  Widget excludeFromSemantics() {
    return ExcludeSemantics(child: this);
  }

  /// Wrap widget as live region for dynamic updates
  Widget asLiveRegion({bool assertive = false}) {
    return Semantics(
      liveRegion: true,
      child: this,
    );
  }
}

File: apps/apple/BayNavigatorCore/Sources/BayNavigatorCore/Services/AccessibilityService.swift
import Foundation
import SwiftUI
#if canImport(UIKit)
import UIKit
#endif

/// WCAG 2.2 AAA Accessibility Service
///
/// Provides accessibility utilities for Bay Navigator including:
/// - Reduced motion detection and preference
/// - High contrast mode support
/// - Screen reader announcements
/// - Text customization (WCAG 3.0 draft)
/// - Semantic label helpers
public final class AccessibilityService: Sendable {
    public static let shared = AccessibilityService()

    // MARK: - UserDefaults Keys

    private let textScaleKey = "accessibility_text_scale"
    private let lineHeightKey = "accessibility_line_height"
    private let letterSpacingKey = "accessibility_letter_spacing"
    private let wordSpacingKey = "accessibility_word_spacing"

    private init() {}

    // MARK: - System Accessibility Settings

    /// Check if reduced motion is preferred (system setting)
    @MainActor
    public var shouldReduceMotion: Bool {
        #if canImport(UIKit) && !os(watchOS)
        return UIAccessibility.isReduceMotionEnabled
        #else
        return false
        #endif
    }

    /// Check if VoiceOver is running
    @MainActor
    public var isVoiceOverRunning: Bool {
        #if canImport(UIKit) && !os(watchOS)
        return UIAccessibility.isVoiceOverRunning
        #else
        return false
        #endif
    }

    /// Check if bold text is enabled
    @MainActor
    public var isBoldTextEnabled: Bool {
        #if canImport(UIKit) && !os(watchOS)
        return UIAccessibility.isBoldTextEnabled
        #else
        return false
        #endif
    }

    /// Check if reduce transparency is enabled
    @MainActor
    public var isReduceTransparencyEnabled: Bool {
        #if canImport(UIKit) && !os(watchOS)
        return UIAccessibility.isReduceTransparencyEnabled
        #else
        return false
        #endif
    }

    /// Check if differentiate without color is enabled
    @MainActor
    public var isDifferentiateWithoutColorEnabled: Bool {
        #if canImport(UIKit) && !os(watchOS)
        return UIAccessibility.shouldDifferentiateWithoutColor
        #else
        return false
        #endif
    }

    // MARK: - Screen Reader Announcements

    /// Announce a message to screen readers
    @MainActor
    public func announce(_ message: String) {
        #if canImport(UIKit) && !os(watchOS)
        UIAccessibility.post(notification: .announcement, argument: message)
        #endif
    }

    /// Announce a message with a slight delay (useful after UI updates)
    @MainActor
    public func announceAfterDelay(_ message: String, delay: TimeInterval = 0.5) {
        Task { @MainActor in
            try? await Task.sleep(nanoseconds: UInt64(delay * 1_000_000_000))
            announce(message)
        }
    }

    /// Announce screen change to VoiceOver
    @MainActor
    public func announceScreenChange(_ message: String? = nil) {
        #if canImport(UIKit) && !os(watchOS)
        UIAccessibility.post(notification: .screenChanged, argument: message)
        #endif
    }

    /// Announce layout change to VoiceOver
    @MainActor
    public func announceLayoutChange(_ element: Any? = nil) {
        #if canImport(UIKit) && !os(watchOS)
        UIAccessibility.post(notification: .layoutChanged, argument: element)
        #endif
    }

    // MARK: - Common Announcements

    /// Announce loading state
    @MainActor
    public func announceLoading() {
        announce("Loading")
    }

    /// Announce loading complete
    @MainActor
    public func announceLoadingComplete(itemCount: Int? = nil) {
        if let count = itemCount {
            announce("Loaded \(count) items")
        } else {
            announce("Loading complete")
        }
    }

    /// Announce error
    @MainActor
    public func announceError(_ error: String) {
        announce("Error: \(error)")
    }

    /// Announce filter change
    @MainActor
    public func announceFilterChange(resultCount: Int) {
        announce("Showing \(resultCount) results")
    }

    /// Announce save/unsave action
    @MainActor
    public func announceSaveAction(isSaved: Bool, programName: String) {
        if isSaved {
            announce("\(programName) saved to your list")
        } else {
            announce("\(programName) removed from your list")
        }
    }

    // MARK: - Animation Helpers

    /// Get animation duration respecting reduced motion
    @MainActor
    public func getAnimationDuration(_ normalDuration: TimeInterval) -> TimeInterval {
        shouldReduceMotion ? 0 : normalDuration
    }

    /// Get animation respecting reduced motion
    @MainActor
    public func getAnimation(_ normalAnimation: Animation?) -> Animation? {
        shouldReduceMotion ? nil : normalAnimation
    }

    // MARK: - Text Settings

    /// Save text customization settings
    public func saveTextSettings(_ settings: TextSettings) {
        UserDefaults.standard.set(settings.textScale, forKey: textScaleKey)
        UserDefaults.standard.set(settings.lineHeight, forKey: lineHeightKey)
        UserDefaults.standard.set(settings.letterSpacing, forKey: letterSpacingKey)
        UserDefaults.standard.set(settings.wordSpacing, forKey: wordSpacingKey)
    }

    /// Load text customization settings
    public func loadTextSettings() -> TextSettings {
        TextSettings(
            textScale: UserDefaults.standard.double(forKey: textScaleKey).nonZeroOrDefault(1.0),
            lineHeight: UserDefaults.standard.double(forKey: lineHeightKey).nonZeroOrDefault(1.5),
            letterSpacing: UserDefaults.standard.double(forKey: letterSpacingKey),
            wordSpacing: UserDefaults.standard.double(forKey: wordSpacingKey)
        )
    }

    /// Reset text settings to defaults
    public func resetTextSettings() {
        UserDefaults.standard.removeObject(forKey: textScaleKey)
        UserDefaults.standard.removeObject(forKey: lineHeightKey)
        UserDefaults.standard.removeObject(forKey: letterSpacingKey)
        UserDefaults.standard.removeObject(forKey: wordSpacingKey)
    }
}

// MARK: - Double Extension

private extension Double {
    func nonZeroOrDefault(_ defaultValue: Double) -> Double {
        self == 0 ? defaultValue : self
    }
}

// MARK: - Text Settings

/// Text customization settings (WCAG 3.0 draft compliance)
public struct TextSettings: Codable, Equatable, Sendable {
    public var textScale: Double      // 0.8 - 1.5
    public var lineHeight: Double     // 1.5 - 2.5
    public var letterSpacing: Double  // 0 - 0.1em
    public var wordSpacing: Double    // 0 - 0.2em

    public init(
        textScale: Double = 1.0,
        lineHeight: Double = 1.5,
        letterSpacing: Double = 0.0,
        wordSpacing: Double = 0.0
    ) {
        self.textScale = textScale
        self.lineHeight = lineHeight
        self.letterSpacing = letterSpacing
        self.wordSpacing = wordSpacing
    }

    public var isDefault: Bool {
        textScale == 1.0 &&
        lineHeight == 1.5 &&
        letterSpacing == 0.0 &&
        wordSpacing == 0.0
    }

    /// Apply text settings to a Font
    public func scaledFont(_ font: Font) -> Font {
        // Note: SwiftUI doesn't support all text styling, so we scale the font
        return font
    }
}

// MARK: - High Contrast Colors

/// High contrast color overrides (10:1+ ratio for WCAG AAA)
public struct HighContrastColors {
    public static let text = Color(red: 0, green: 0, blue: 0)
    public static let textOnDark = Color(red: 1, green: 1, blue: 1)
    public static let background = Color(red: 1, green: 1, blue: 1)
    public static let backgroundDark = Color(red: 0, green: 0, blue: 0)
    public static let primary = Color(red: 0, green: 0.25, blue: 0.25)       // Darker teal for 10:1
    public static let primaryOnDark = Color(red: 0.5, green: 1, blue: 1)     // Bright cyan
    public static let link = Color(red: 0, green: 0, blue: 0.8)              // Traditional link blue
    public static let linkOnDark = Color(red: 0.6, green: 0.8, blue: 1)
    public static let focus = Color(red: 0, green: 0, blue: 0)
    public static let focusOnDark = Color(red: 1, green: 1, blue: 0)         // Yellow focus ring
    public static let error = Color(red: 0.8, green: 0, blue: 0)
    public static let errorOnDark = Color(red: 1, green: 0.4, blue: 0.4)
    public static let success = Color(red: 0, green: 0.4, blue: 0)
    public static let successOnDark = Color(red: 0.4, green: 1, blue: 0.4)
}

// MARK: - Semantic Labels

/// Semantic label helpers for consistent accessibility descriptions
public enum SemanticLabels {
    // Navigation
    public static let home = "Home"
    public static let directory = "Program Directory"
    public static let saved = "Saved Programs"
    public static let settings = "Settings"
    public static let search = "Search programs"
    public static let filter = "Filter programs"
    public static let sort = "Sort programs"

    // Actions
    public static func saveProgram(_ name: String) -> String {
        "Save \(name) to your list"
    }

    public static func unsaveProgram(_ name: String) -> String {
        "Remove \(name) from your list"
    }

    public static func shareProgram(_ name: String) -> String {
        "Share \(name)"
    }

    public static func callPhone(_ phone: String) -> String {
        "Call \(phone)"
    }

    public static func openWebsite(_ name: String) -> String {
        "Open \(name) website"
    }

    public static func getDirections(_ location: String) -> String {
        "Get directions to \(location)"
    }

    // States
    public static func programSaved(_ name: String) -> String {
        "\(name) is saved"
    }

    public static func programNotSaved(_ name: String) -> String {
        "\(name) is not saved"
    }

    public static func filterActive(_ count: Int) -> String {
        "\(count) filters active"
    }

    public static func resultsCount(_ count: Int) -> String {
        "\(count) programs found"
    }

    public static let loadingPrograms = "Loading programs"

    // Categories
    public static func categoryLabel(_ category: String) -> String {
        "\(category) programs"
    }

    // Groups
    public static let groupLabels: [String: String] = [
        "seniors": "Seniors 65 and older",
        "veterans": "Veterans and military families",
        "disabled": "People with disabilities",
        "lowincome": "Low income individuals and families",
        "students": "Students",
        "families": "Families with children",
        "unemployed": "Unemployed or job seekers",
        "homeless": "People experiencing homelessness",
        "immigrants": "Immigrants and refugees",
        "youth": "Youth and young adults"
    ]

    public static func getGroupLabel(_ groupId: String) -> String {
        groupLabels[groupId.lowercased()] ?? groupId
    }
}

// MARK: - SwiftUI View Modifiers

/// Accessible animation modifier that respects reduced motion
public struct AccessibleAnimation: ViewModifier {
    @Environment(\.accessibilityReduceMotion) var reduceMotion
    let animation: Animation?

    public func body(content: Content) -> some View {
        content.animation(reduceMotion ? nil : animation, value: UUID())
    }
}

/// Accessible opacity modifier
public struct AccessibleOpacity: ViewModifier {
    @Environment(\.accessibilityReduceMotion) var reduceMotion
    let opacity: Double
    let animation: Animation?

    public func body(content: Content) -> some View {
        if reduceMotion {
            content.opacity(opacity)
        } else {
            content
                .opacity(opacity)
                .animation(animation, value: opacity)
        }
    }
}

// MARK: - View Extensions

public extension View {
    /// Apply animation respecting reduced motion preference
    func accessibleAnimation(_ animation: Animation? = .default) -> some View {
        modifier(AccessibleAnimation(animation: animation))
    }

    /// Apply opacity with animation respecting reduced motion
    func accessibleOpacity(_ opacity: Double, animation: Animation? = .easeInOut) -> some View {
        modifier(AccessibleOpacity(opacity: opacity, animation: animation))
    }

    /// Add semantic label for accessibility
    func accessibilitySemanticLabel(_ label: String) -> some View {
        self.accessibilityLabel(label)
    }

    /// Mark as a button with label
    func accessibleButton(_ label: String, hint: String? = nil) -> some View {
        self
            .accessibilityLabel(label)
            .accessibilityHint(hint ?? "")
            .accessibilityAddTraits(.isButton)
    }

    /// Mark as a heading
    func accessibleHeading(_ label: String) -> some View {
        self
            .accessibilityLabel(label)
            .accessibilityAddTraits(.isHeader)
    }

    /// Mark as an image with description
    func accessibleImage(_ description: String) -> some View {
        self
            .accessibilityLabel(description)
            .accessibilityAddTraits(.isImage)
    }

    /// Mark as a link
    func accessibleLink(_ label: String) -> some View {
        self
            .accessibilityLabel(label)
            .accessibilityAddTraits(.isLink)
    }

    /// Hide from accessibility (for decorative elements)
    func accessibilityDecorative() -> some View {
        self.accessibilityHidden(true)
    }

    /// Mark as a live region for dynamic updates
    func accessibilityLiveRegion() -> some View {
        self.accessibilityAddTraits(.updatesFrequently)
    }
}
